### PR TITLE
[System Settings] Use LocationAwareConfigRepository

### LINKED ISSUE #14373: [System Settings] Use LocationAwareConfigRepository
This allows to store the system config into the database or any other supported location. 
Approach similar to what we do e.g. with `\Pimcore\CustomView\Config` or `\Pimcore\Perspective\Config`. 

Please consider #14372 which should be done first.

### PR DESCRIPTION
**Pull request by lukmzig**
## Changes in this pull request  
Resolves #14373

## Additional info  
We removed system.yaml as it will be not used anymore
We added 2 new configs for System settings and Admin system settings (branding).
We split system settings into 2 separate settings views as they are using different config classes
We read the system settings for relevant data from the new Config classes.

### CODE REVIEW

File: bundles/AdminBundle/src/Helper/SystemConfig.php [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+use Pimcore\Config\LocationAwareConfigRepository;
+
+class SystemConfig
+{
+    public static function get(LocationAwareConfigRepository $repository, string $key): array
+    {
+        $config = self::getConfigDataByKey($repository, $key);
+
+        return $config;
```
Comment (kingjia90) [inline bundles/AdminBundle/src/Helper/SystemConfig.php]:
```suggestion
        return self::getConfigDataByKey($repository, $key);
```

File: bundles/AdminBundle/public/js/pimcore/settings/systemAppearance.js [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+                this.getTabPanel();
+
+            }.bind(this)
+        });
+    },
+
+    getValue: function (key, ignoreCheck) {
+
+        var nk = key.split("\.");
+        var current = this.data.values;
```
Comment (robertSt7) [inline bundles/AdminBundle/public/js/pimcore/settings/systemAppearance.js]:
```suggestion
        const nk = key.split("\.");
        let current = this.data.values;
```

File: bundles/AdminBundle/public/js/pimcore/settings/systemAppearance.js [hunk 3]
```diff
@@ -0,0 +1,288 @@
...
+                                html: '<img src="'+Routing.generate('pimcore_settings_display_custom_logo')+'" />',
+                            }, {
+                                xtype: "button",
+                                text: t("upload"),
+                                iconCls: "pimcore_icon_upload",
+                                handler: function () {
+                                    pimcore.helpers.uploadDialog(Routing.generate('pimcore_admin_settings_uploadcustomlogo'), null,
+                                        function () {
+                                            var cont = Ext.getCmp("pimcore_custom_branding_logo");
+                                            var date = new Date();
```
Comment (robertSt7) [inline bundles/AdminBundle/public/js/pimcore/settings/systemAppearance.js]:
```suggestion
                                            const cont = Ext.getCmp("pimcore_custom_branding_logo");
                                            const date = new Date();
```

File: bundles/AdminBundle/public/js/pimcore/settings/systemAppearance.js [hunk 4]
```diff
@@ -0,0 +1,288 @@
...
+                                xtype: "button",
+                                text: t("delete"),
+                                iconCls: "pimcore_icon_delete",
+                                handler: function () {
+                                    Ext.Ajax.request({
+                                        url: Routing.generate('pimcore_admin_settings_deletecustomlogo'),
+                                        method: "DELETE",
+                                        success: function (response) {
+                                            var cont = Ext.getCmp("pimcore_custom_branding_logo");
+                                            var date = new Date();
```
Comment (robertSt7) [inline bundles/AdminBundle/public/js/pimcore/settings/systemAppearance.js]:
```suggestion
                                            const cont = Ext.getCmp("pimcore_custom_branding_logo");
                                            const date = new Date();
```

File: bundles/AdminBundle/public/js/pimcore/settings/systemAppearance.js [hunk 5]
```diff
@@ -0,0 +1,288 @@
...
+                                    urlToCustomImageField.setValue('');
+                                }
+                            }]
+                        }
+                ]
+            });
+
+            this.panel.add(this.layout);
+
+            var tabPanel = Ext.getCmp("pimcore_panel_tabs");
```
Comment (robertSt7) [inline bundles/AdminBundle/public/js/pimcore/settings/systemAppearance.js]:
```suggestion
            const tabPanel = Ext.getCmp("pimcore_panel_tabs");
```

File: bundles/AdminBundle/public/js/pimcore/settings/systemAppearance.js [hunk 6]
```diff
@@ -0,0 +1,288 @@
...
+            tabPanel.setActiveItem(this.panel);
+
+            pimcore.layout.refresh();
+        }
+
+        return this.panel;
+    },
+
+    activate: function () {
+        var tabPanel = Ext.getCmp("pimcore_panel_tabs");
```
Comment (robertSt7) [inline bundles/AdminBundle/public/js/pimcore/settings/systemAppearance.js]:
```suggestion
        const tabPanel = Ext.getCmp("pimcore_panel_tabs");
```

File: bundles/AdminBundle/public/js/pimcore/settings/systemAppearance.js [hunk 7]
```diff
@@ -0,0 +1,288 @@
...
+        tabPanel.setActiveItem("pimcore_settings_system_appearance");
+    },
+
+    save: function () {
+
+        this.layout.mask();
+
+        var values = this.layout.getForm().getFieldValues();
```
Comment (robertSt7) [inline bundles/AdminBundle/public/js/pimcore/settings/systemAppearance.js]:
```suggestion
        const values = this.layout.getForm().getFieldValues();
```

File: bundles/AdminBundle/public/js/pimcore/settings/systemAppearance.js [hunk 8]
```diff
@@ -0,0 +1,288 @@
...
+            method: "PUT",
+            params: {
+                data: Ext.encode(values)
+            },
+            success: function (response) {
+
+                this.layout.unmask();
+
+                try {
+                    var res = Ext.decode(response.responseText);
```
Comment (robertSt7) [inline bundles/AdminBundle/public/js/pimcore/settings/systemAppearance.js]:
```suggestion
                    const res = Ext.decode(response.responseText);
```

File: bundles/AdminBundle/src/DependencyInjection/Configuration.php [hunk 9]
```diff
@@ -135,6 +135,7 @@ public function getConfigTreeBuilder(): TreeBuilder
         ;
 
         $this->addAdminNode($rootNode);
+        ConfigurationHelper::addConfigLocationWithWriteTargetNodes($rootNode, ['admin_system_settings' => '/var/config/admin_system_settings']);
```
Comment (robertSt7) [inline bundles/AdminBundle/src/DependencyInjection/Configuration.php]:
```suggestion
```

File: bundles/AdminBundle/src/DependencyInjection/PimcoreAdminExtension.php [hunk 10]
```diff
@@ -75,6 +76,19 @@ public function load(array $configs, ContainerBuilder $container): void
...
+            $container,
+            new FileLocator($configDir)
+        );
+
+        //load configs
+        $configs = ConfigurationHelper::getSymfonyConfigFiles($configDir);
+        foreach ($configs as $config) {
+            $configLoader->load($config);
+        }
+
```
Comment (robertSt7) [inline bundles/AdminBundle/src/DependencyInjection/PimcoreAdminExtension.php]: Can we use `LocationAwareConfigRepository::loadSymfonyConfigFiles` here?

File: doc/21_Deployment/03_Configuration_Environments.md [hunk 11]
```diff
@@ -97,6 +103,8 @@ pimcore:
 	          type: 'symfony-config'
               options:
                 directory: '/var/www/html/var/config/staticroutes'
+              read_target:
+                type: 'settings-store'
```
Comment (robertSt7) [inline doc/21_Deployment/03_Configuration_Environments.md]:
```suggestion
```

File: doc/21_Deployment/03_Configuration_Environments.md [hunk 12]
```diff
@@ -69,6 +69,8 @@ pimcore:
         custom_reports:
             write_target:
 	          type: 'settings-store'
+	        read_target:
+              type: 'symfony-config'
```
Comment (robertSt7) [inline doc/21_Deployment/03_Configuration_Environments.md]:
```suggestion
```

**Review by robertSt7 — APPROVED** [review]

File: doc/23_Installation_and_Upgrade/09_Upgrade_Notes/README.md [hunk 13]
```diff
@@ -308,6 +308,28 @@ pimcore:
...
+  To save these settings into the settings store, please add following to your configuration:
+```yaml
+pimcore_admin:
+    config_location:
+        admin_system_settings:
+            write_target:
+                type: 'settings-store'
+            read_target:
+                type: 'settings-store'
+```
```
Comment (robertSt7) [inline doc/23_Installation_and_Upgrade/09_Upgrade_Notes/README.md]:
```suggestion
- [System Settings] System Settings are now implementing the LocationAwareConfigRepository. See [preparing guide](../07_Updating_Pimcore/11_Preparing_for_V11.md)
```
```

File: bundles/AdminBundle/public/js/pimcore/settings/systemAppearance.js [hunk 14]
```diff
@@ -0,0 +1,288 @@
...
+        this.getData();
+    },
+
+    getData: function () {
+        Ext.Ajax.request({
+            url: Routing.generate('pimcore_appearance_admin_settings_get'),
+            success: function (response) {
+
+                this.data = Ext.decode(response.responseText);
+                console.log(this.data);
```
Comment (kingjia90) [inline bundles/AdminBundle/public/js/pimcore/settings/systemAppearance.js]:
```suggestion
```

File: doc/21_Deployment/03_Configuration_Environments.md [hunk 15]
```diff
@@ -84,6 +86,10 @@ pimcore:
               type: 'symfony-config'
               options:
                 directory: '/var/www/html/var/config/web_to_print'
+            read_target:
+              type: 'symfony-config'
+              options:
+                directory: '/var/www/html/var/config/web_to_print'
```
Comment (kingjia90) [inline doc/21_Deployment/03_Configuration_Environments.md]:
```suggestion
```

duplicated lines of 
https://github.com/pimcore/pimcore/pull/14867/files?w=1#diff-d39e3f0461133fdc79416157bdcc5f5f98cacecae0a12b90f1385cdaf1ca8855R85-R88

Comment (kingjia90) [reply doc/21_Deployment/03_Configuration_Environments.md]: but looking at some change requested by robert, even line 85-88 should be removed

File: bundles/AdminBundle/src/DependencyInjection/PimcoreAdminExtension.php (hunk 10 above)
Comment (mcop1) [reply bundles/AdminBundle/src/DependencyInjection/PimcoreAdminExtension.php]: We can and we should now :)

File: bundles/AdminBundle/src/DependencyInjection/PimcoreAdminExtension.php [hunk 16]
```diff
@@ -16,6 +16,8 @@
 
 namespace Pimcore\Bundle\AdminBundle\DependencyInjection;
 
+use Pimcore\Bundle\CoreBundle\DependencyInjection\ConfigurationHelper;
```
Comment (robertSt7) [inline bundles/AdminBundle/src/DependencyInjection/PimcoreAdminExtension.php]:
```suggestion
```

File: bundles/AdminBundle/public/js/pimcore/settings/systemAppearance.js [hunk 17]
```diff
@@ -0,0 +1,287 @@
...
+            }.bind(this)
+        });
+    },
+
+    getValue: function (key, ignoreCheck) {
+
+        const nk = key.split("\.");
+        let current = this.data.values;
+
+        for (var i = 0; i < nk.length; i++) {
```
Comment (robertSt7) [inline bundles/AdminBundle/public/js/pimcore/settings/systemAppearance.js]:
```suggestion
        for (let i = 0; i < nk.length; i++) {
```

**Pull request merged**